sklearn/compose/_column_transformer.py

```diff
@@ -630,11 +630,11 @@ def _get_column_indices(X, key):
 
     if _check_key_type(key, int):
```

Comment:
We could simplify this code to:
```
if _check_key_type(key, str):
    ...
else:
    return np.atleast_1d(np.arange(n_columns)[key]).tolist()
```
(maybe with a check that it's also at most 1d)